x/records/keeper/callback_lsm_transfer_test.go

```diff
@@ -0,0 +1,112 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	_ "github.com/stretchr/testify/suite"
+
+	"github.com/golang/protobuf/proto" //nolint:staticcheck
+
+	transfertypes "github.com/cosmos/ibc-go/v5/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v5/modules/core/04-channel/types"
+	ibctesting "github.com/cosmos/ibc-go/v5/testing"
+
+	icacallbackstypes "github.com/Stride-Labs/stride/v9/x/icacallbacks/types"
+	recordskeeper "github.com/Stride-Labs/stride/v9/x/records/keeper"
+	"github.com/Stride-Labs/stride/v9/x/records/types"
+)
+
+var (
+	LSMTokenDenom = "cosmosvaloperxxx/42"
+)
+
+func (s *KeeperTestSuite) SetupLSMTransferCallback() []byte {
+	delegationAccountOwner := fmt.Sprintf("%s.%s", HostChainId, "DELEGATION")
+	s.CreateICAChannel(delegationAccountOwner)
+	delegationAddress := s.IcaAddresses[delegationAccountOwner]
+
+	// we need a valid ibc denom here or the transfer will fail
+	prefixedDenom := transfertypes.GetPrefixedDenom(transfertypes.PortID, ibctesting.FirstChannelID, LSMTokenDenom)
+	denomTrace := transfertypes.ParseDenomTrace(prefixedDenom)
+	ibcDenom := denomTrace.IBCDenom()
+	s.App.TransferKeeper.SetDenomTrace(s.Ctx, denomTrace)
+
+	deposit := types.LSMTokenDeposit{
+		ChainId:  HostChainId,
+		Denom:    LSMTokenDenom,
+		IbcDenom: ibcDenom,
+		Status:   types.LSMTokenDeposit_TRANSFER_IN_PROGRESS,
+	}
+	s.App.RecordsKeeper.SetLSMTokenDeposit(s.Ctx, deposit)
+
+	callbackArgs := types.TransferLSMTokenCallback{
+		Deposit:                      &deposit,
+		TransferChannelId:            ibctesting.FirstChannelID,
+		HostZoneDepositAddress:       s.TestAccs[0].String(),
+		HostZoneDelegationIcaAddress: delegationAddress,
+	}
+	callbackArgsBz, err := proto.Marshal(&callbackArgs)
+	s.Require().NoError(err, "no error expected when marshalling callback args")
+
+	return callbackArgsBz
+}
+
+func (s *KeeperTestSuite) TestLSMTransferCallback_Successful() {
+	callbackArgsBz := s.SetupLSMTransferCallback()
+
+	// Call the callback with a successful response
+	ackSuccess := &icacallbackstypes.AcknowledgementResponse{
+		Status: icacallbackstypes.AckResponseStatus_SUCCESS,
+	}
+	err := recordskeeper.LSMTransferCallback(s.App.RecordsKeeper, s.Ctx, channeltypes.Packet{}, ackSuccess, callbackArgsBz)
+	s.Require().NoError(err, "no error expected when executing callback")
+
+	// Confirm deposit has been updated to DETOKENIZATION_QUEUE
+	record, found := s.App.RecordsKeeper.GetLSMTokenDeposit(s.Ctx, HostChainId, LSMTokenDenom)
+	s.Require().True(found, "deposit found")
```

Comment:
Doesn't the string as the second input only get used if the require fails? So I normally phrase these the opposite way "deposit not found!" because that is the case when I see the string

Reply:
I get that you are using these as labels as to which test triggered it, but then it will print "deposit found" when the deposit was in fact not found.... seems confusing

Reply:
The way it prints out is:
```
expected: X
actual: Y
{description}
```

So if you had "deposit not found", it would print out as 
```
expected: True
actual: False
deposit not found
```
Which is a bit confusing cause it makes it look like you're expecting "deposit not found" to be true.

It probably doesn't hurt to just be explicit though since both of these are confusing. I'll update to "deposit should have been found but was not"